test/unit/cpu/test-indexsets.cxx

```diff
@@ -0,0 +1,32 @@
+#include "gtest/gtest.h"
+
+#include "buildIndexSet.hxx"
+
+#include "RAJA/RAJA.hxx"
+
+class IndexSetTest : public ::testing::Test {
+  protected:
+    virtual void SetUp() {
+      for (unsigned ibuild = 0; ibuild < NumBuildMethods; ++ibuild) {
+        buildIndexSet(index_sets_, static_cast<IndexSetBuildMethod>(ibuild));
+      }
+    }
+
+    RAJA::IndexSet index_sets_[NumBuildMethods];
+
+};
+
+TEST_F(IndexSetTest, IndexSetEquality) {
+  for (unsigned ibuild = 1; ibuild < NumBuildMethods; ++ibuild) {
+    EXPECT_EQ(index_sets_[ibuild], index_sets_[1]);
+  }
+}
+
+// TODO: tests for adding other invalid types
```

Comment:
Eventually, we want to have an IndexSet model that supports arbitrary segment types without the switch-yard crap. So we may not need this in the long run, but is OK for now.